nixos/modules/programs/fish.nix

```
@@ -53,7 +55,24 @@ in
         type = lib.types.bool;
       };
 
-      package = lib.mkPackageOption pkgs "fish" { };
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.fishWrapper;
+        defaultText = lib.literalExpression "pkgs.fish";
+        example = lib.literalExpression ''
+          pkgs.fishWrapper.override {
+            pluginPkgs = with pkgs.fishPlugins; [
+              autopair
+              fifc
+              done
+            ];
+          }
+        '';
+        apply = fishWrapper: fishWrapper.override { pluginPkgs = cfg.plugins; };
```

```suggestion
        apply = fishWrapper: fishWrapper.override { pluginPkgs = cfg.plugins; };
```

What  if user set`cfg.package` as an unwrapped fish? What if user set `cfg.package` as an already overrided fishWrapper?